clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -623,15 +623,15 @@ else if (value instanceof String)
                     } else if (value instanceof String) {
                         return Integer.parseInt(trimmed);
                     } else {
-                        throw new ConfigException(name, value, "Expected value to be an number.");
+                        throw new ConfigException(name, value, "Expected value to be an integer, but it was a " + value.getClass().getName());
                     }
                 case SHORT:
                     if (value instanceof Short) {
                         return (Short) value;
                     } else if (value instanceof String) {
                         return Short.parseShort(trimmed);
                     } else {
-                        throw new ConfigException(name, value, "Expected value to be an number.");
+                        throw new ConfigException(name, value, "Expected value to be a short / string, but it was a " + value.getClass().getName());
```

Comment:
Also, `short` is not so clear for people who are just using configs. Would 16-bit integer be clearer?